File: Json Parsing 2/Json Parsing 2/Objective-c Files/DatabaseHelper.m
//
//  DatabaseHelper.m
//  Json Parsing 2
//
//  Created by Mac on 14/09/20.
//  Copyright © 2020 Mac. All rights reserved.
//

#import "DatabaseHelper.h"
#import <CoreData/CoreData.h>
#import "Json_Parsing_2-Swift.h"
#import "Person+CoreDataClass.h"
#import "SecondVC.h"

@interface DatabaseHelper ()

@end

@implementation DatabaseHelper

@synthesize appDelegate, persistentContainer;

+ (id)sharedInstance {
    static DatabaseHelper *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (AppDelegate *)appDelegate{
    return (AppDelegate *)[[UIApplication sharedApplication]delegate];
}

- (NSManagedObjectContext *)managedObjectContext{
    return [[self appDelegate] persistentContainer].viewContext;
}

    -(NSArray *) getPersondb {

    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:@"Person"];
    NSArray *result  = [self.managedObjectContext executeFetchRequest: request error:nil];
    return result;
}

    -(void) saveData:(NSDictionary *) object{
        
    Person *person = [NSEntityDescription insertNewObjectForEntityForName: @"Person" inManagedObjectContext: [self managedObjectContext]];

    person.pid  = [object objectForKey:@"id"];
    person.name = [object objectForKey:@"name"];
    person.email = [object objectForKey:@"email"];
    person.address = [object objectForKey:@"address"];
    person.gender = [object objectForKey:@"gender"];


        NSError *error = nil;

        if ([[self managedObjectContext] save:&error] == NO)
        {
        NSAssert(NO, @"Error saving context: %@\n%@", [error localizedDescription], [error userInfo]);
        }

   }

@end

File: Json Parsing 2/Json Parsing 2/Swifts Files/DataBaseHelper.swift
//
//  DataBaseHelper.swift
//  Json Parsing 2
//
//  Created by Mac on 09/09/20.
//  Copyright © 2020 Mac. All rights reserved.
//

import Foundation
import CoreData
import UIKit

class DataBaseHelper{
    
    static var shareInstance = DataBaseHelper()
    let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
    
    func save(object: jsonstruct) {
        
        let countries =  NSEntityDescription.insertNewObject(forEntityName: "Countries", into: context!) as! Countries
        countries.name = object.name
        countries.capital = object.capital
        
        do{
            try context?.save()
        }catch{
            print("Error in saving data")
            
        }
    }
}
